File: leetcode solutions/C++/palindrome permutation/palindromePermutation.cpp
#include <iostream>
#include <set>

using namespace std;

class Solution {
public:
    bool canPermutePalindrome(string s) {

        if (s.length() == 0) {
            return false;
        }
        set<char> chSet;

        for (int i =0; i < s.length(); i++) {
            if (chSet.find(s[i])==chSet.end())
                chSet.insert(s[i]);
            else
                chSet.erase(s[i]);
        }

        if (s.length() % 2 == 0)
            return (chSet.size() == 0);
        else
            return (chSet.size() == 1);
    }
};

File: leetcode solutions/Python/palindrome permutation/palindromePermutation.py
class Solution(object):
    def canPermutePalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        if not s:
            return False

        frequencies = set()

        for ch in s:
            if ch in frequencies:
                frequencies.remove(ch)
            else:
                frequencies.add(ch)

        if len(s) % 2 == 0:
            return len(frequencies) == 0
        else:
            return len(frequencies) == 1
